drivers/lora/sx12xx_common.c

```diff
@@ -78,15 +80,25 @@ int sx12xx_lora_send(const struct device *dev, uint8_t *data,
 int sx12xx_lora_recv(const struct device *dev, uint8_t *data, uint8_t size,
 		     k_timeout_t timeout, int16_t *rssi, int8_t *snr)
 {
+	struct k_poll_signal done;
```

Comment:
Static is not a solution because if we can't guarantee that done is only ever raised while `sx12xx_lora_recv` is running we also can't guarantee that the output parameters are only written while the function is running. We need to ensure that as those pointers can also be stack variables.

I "think" that the current implementation makes the necessary guarantees,  which I have tried to explain in the `sx12xx_ev_rx_done` callback.

Reply:
See also the commit message from this change:
> Transition the receive synchronisation method from a single driver
semaphore to a function specific k_poll_signal. This is required to
allow the modem to be released without introducing race conditions on
the signalling mechanism.